### DISCUSSION TITLE
Support Linking Execution Trace to DatasetItem without Fetching Entire Dataset

### DISCUSSION THREAD

**Discussion by sfarthin** [langfuse/langfuse#1862] [Ideas]
### Describe the feature or potential improvement

We need a method to efficiently link an execution trace to a specific DatasetItem without the need to fetch the entire dataset first. Currently, the process requires retrieving a full dataset just to access a single Dataset Item, which is not optimal for performance.

The Langfuse API currently supports creating a dataset run item and linking it with a traceId as described here: [Langfuse API - Create Dataset Run Item](https://api.reference.langfuse.com/#post-/api/public/dataset-run-items). However, to use the link method from the SDK, one must first fetch an entire dataset to obtain a single Dataset Item.

# Proposal:

Develop a feature or method within the SDK/API that allows linking a trace directly to a Dataset Item by ID, without the necessity of loading the entire dataset. 

```js
const datasetItem = await langfuseClient.getDatasetItem(datasetItem.id);
datasetItem.link(trace, 'some_run_name');
```

This feature would be particularly useful in scenarios where users are dealing with large datasets and need to perform operations on individual items without the overhead of loading entire datasets.

**Comment by marcklingen** [comment]
great idea, turning this into an issue: https://github.com/langfuse/langfuse/issues/1863
